### PR TITLE
vala: use grammar from @vala-lang

### PR DESCRIPTION
**Pull request by Prince781**
Use the grammar from https://github.com/vala-lang/tree-sitter-vala

This grammar seems to work better on more inputs and is maintained more frequently.

### CODE REVIEW

**Comment by clason** [comment]
Be aware that changing parsers for a supported language with incompatible queries is a massively breaking change for users. This should not be done lightly.

**Comment by Prince781** [comment]
@clason sure, but the current parser for Vala doesn't seem to [work](https://github.com/vala-lang/tree-sitter-vala/issues/1#issuecomment-1083987163), so I don't think this will affect users much.

**Comment by Prince781** [comment]
@clason Do I need generating from grammar if I already provide `src/parser.c`?

**Comment by clason** [comment]
Sorry, that issue is a bit vague to conclude that. The old vala parser is tested by and passes our CI on Neovim 0.7, so something else must be going on there.

**Comment by clason** [comment]
> @clason Do I need generating from grammar if I already provide `src/parser.c`?

No.

**Comment by clason** [comment]
@matbme

**Comment by clason** [comment]
The highlights are using an invalid capture name; the only currently allowed captures in nvim-treesitter are those listed in https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#highlights

**Comment by Prince781** [comment]
@clason sorry, should be fixed

**Comment by clason** [comment]
I still will not merge this without a (sufficiently clear) explanation of why your parser is significantly and objectively better than the old one.

**Comment by theHamsta** [comment]
The old one didn't really work great (couldn't parse many constructs) so I don't think we break to many users' experiences. Can't tell about the new one I guess one just needs to try it out.

**Comment by clason** [comment]
Would still like to see _some_ before/after example (and would like to hear from the previous maintainer), just on principle...

(This will become less of an issue when `maintained` is removed, but while this is still around in the wild, it will affect unsuspecting users.)

**Comment by theHamsta** [comment]
You could also quickly fix the bug about changing parsers by including the parser URL in the lockfile.

**Comment by clason** [comment]
> You could also quickly fix the bug about changing parsers by including the parser URL in the lockfile.

That would be good in any case, but here the queries seem to be incompatible, which is the actual breaking change I am worried about.

**Comment by theHamsta** [comment]
@matbme

**Comment by clason** [comment]
Already pinged them yesterday. I'd give them a few more days (it is a holiday for many people, after all), but no response would be sufficient reason to switch.

**Comment by matbme** [comment]
I was aware of the [duplication of effort](https://github.com/matbme/tree-sitter-vala/issues/2), but it's interesting to see this new parser in a group with other similar projects. It's true I didn't have a lot of time lately to improve the parser, so I could agree on a change.

Looking at the proposed parser, I see that some things such as some preproc statements and verbatim are still not implemented, which I could gladly help add if that's ok with @Prince781.

**Comment by theHamsta** [comment]
We can still wait with the change. @Prince781, @matbme it would be good to see a complete parser with maybe combined efforts.

A good indication is that a parser can parse large repos of the language without errors https://github.com/theHamsta/tree-sitter-commonlisp/blob/c7e814975ab0d0d04333d1f32391c41180c58919/.github/workflows/ci.yml#L21-L31

**Comment by clason** [comment]
I'm converting this to draft; feel free to reopen (and rebase) when you have combined efforts on whichever parser you decide on!

(By that time, we should have the infrastructure in place to handle switching parser URLs transparently to the user.)

**Comment by matbme** [comment]
@clason I think It's time to transition Vala support to the new repo. I don't have the time to fix everything in the current one and @Prince781's implementation is more flexible.

**Comment by clason** [comment]
@matbme If you approve this change, that's fine with me. But please squash your commits and mark the change as breaking (by using the commit message `feat(vala)!: switch to vala-lang parser`).

@theHamsta should take a look at the new queries.

I would appreciate it if you could look at other downstream plugins (especially `tree-sitter-textobjects`) and check if they have vala queries that need to be adapted (and help them to do so).

File: queries/vala/highlights.scm
Comment (theHamsta) [inline queries/vala/highlights.scm:123]: I think some of those keywords fall under `@exception` or `@repeat`/`@conditional`

Comment (Prince781) [reply queries/vala/highlights.scm:115]: Hmmm...it looks like there were some duplicate entries for some of these keywords.

File: queries/vala/highlights.scm
Comment (theHamsta) [inline queries/vala/highlights.scm:54]: didn't know that our `#match?` would look at more than one argument

**Review by theHamsta — APPROVED** [review]
This looks good. Are we ready to merge?

Also this is the first PR that makes `storageclass` and `type.qualifier` canoncial

**Comment by Prince781** [comment]
Yes, I think it's ready to merge.

**Pull request merged**
